src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -731,21 +745,103 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
-    private void writeServerEnvProperties(File file, Properties properties, Properties projectProperties) throws IOException {
-        Properties combinedEnvProperties = null
+    private void handleServerEnvFileAndProperties(String serverEnvPath, String serverDirectory) {
+        File envFile = new File(serverDirectory, "server.env")
+        Properties configuredProps = combineServerEnvProperties(server.env, envProjectProps);
+
+        if(server.appendServerEnv) {
+            setServerEnvWithAppendServerEnvHelper(envFile, serverEnvPath, configuredProps)
+        }
+        else {
+            setServerEnvHelper(envFile, serverEnvPath, configuredProps)
+        }
+    }
+
+    private void setServerEnvWithAppendServerEnvHelper(File envFile, String serverEnvPath, Properties configuredProps) {
+        Properties serverEnvProps = convertServerEnvToProperties(envFile);
+        Properties mergedProperties = new Properties();
+
+        if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
+            if (serverEnvPath != null) {
+                logger.warn("The " + serverEnvPath + " file is merged with the " + server.serverEnvFile.getCanonicalPath() + " file.")
```

Comment:
Need to set and return serverEnvPath to indicate what happened in this method. I think this would be more accurate to say "The serverEnvFile "+ server.serverEnvFile.getCanonicalPath()+" is merged with the "+serverEnvPath+" file.".

Reply:
Also, even when serverEnvPath is null, we are still merging with the default server.env. So a log message would be good in that case too.